doc/contribute/bin_blobs.rst

```diff
@@ -0,0 +1,229 @@
+reasons on a case by case basis (see library-specific requirements below).
+In case of disagreement, the TSC is the arbiter of whether a particular blob
+fits in one of the above types.
+
+Precompiled library-specific requirements
+=========================================
+
+This section contains additional requirements specific to precompiled library
+blobs.
+
+Any person who wishes to submit a precompiled library must represent that it
+meets these requirements. The project may remove a blob from the upstream
+distribution if it is discovered that the blob fails to meet these requirements
+later on.
+
+Interface header files
+----------------------
+
+The precompiled library must be accompanied by one or more header files,
+distributed under a non-copyleft OSI approved license, that define the interface
+to the library.
+
+Allowed dependencies
+--------------------
+
+This section defines requirements related to external symbols that a library
+blob requires the build system to provide.
+
+* The blob must not depend on Zephyr APIs directly. In other words, it must have
+  been possible to build the binary without any Zephyr source code present at
+  all. This is required for loose coupling and maintainability, since Zephyr
+  APIs may change and such blobs cannot be modified by all project maintainers
+* Instead, if the code in the precompiled library requires functionality
+  provided by Zephyr (or an RTOS in general), an implementation of an OS
+  abstraction layer (aka porting layer) can be provided alongside the library.
+  The implementation of this OS abstraction layer must be in source code form,
+  released under an OSI approved license and thoroughfully documented using
+  Doxygen
+
+Toolchain requirements
+----------------------
+
+Precompiled library blobs must be in a data format which is compatible with and
+can be linked by a toolchain supported by the Zephyr Project (i.e. a toolchain
+that can build Zephyr applications). This is required for maintainability and
+usability. Use of such libraries may require special compiler and/or linker
+flags, however. For example, a porting layer may require special flags, or a
+static archive may require use of specific linker flags.
+
+Limited scope
+-------------
+
+Allowing arbitrary library blobs carries a risk of degrading the degree to
+which the upstream Zephyr software distribution is open source. As an extreme
+example, a target with a zephyr kernel clock driver that is just a porting layer
+around a library blob would not be bootable with open source software.
+
+To mitigate this risk, the scope of upstream library blobs is limited. The
+project maintainers define an open source test suite that an upstream
+target must be able to pass using only open source software included in the
+mainline distribution and its modules. The open source test suite currently
+consists of:
+
+- :file:`samples/philosophers`
+- :file:`tests/kernel`
```

Comment:
I guess we have to start somewhere, but I think future work needs to make it clearer how to run and collect test results. cc @PerMac